src/status_im/contexts/communities/events.cljs

```diff
@@ -64,33 +64,56 @@
       (update :chats <-chats-rpc)
       (update :token-permissions seq)
       (update :categories <-categories-rpc)
+      (assoc :membership-permissions?
+             (some #(= (:type %) constants/community-token-permission-become-member)
+                   (vals (:tokenPermissions c))))
       (assoc :token-images
              (reduce (fn [acc {sym :symbol image :image}]
                        (assoc acc sym image))
                      {}
                      (:communityTokensMetadata c)))))
 
-(rf/reg-event-fx :communities/handle-community
- (fn [{:keys [db]}
-      [community-js]]
-   (when community-js
-     (let [{:keys [token-permissions
-                   token-permissions-check joined id]
-            :as   community} (<-rpc community-js)
-           has-channel-perm? (fn [id-perm-tuple]
-                               (let [{:keys [type]} (second id-perm-tuple)]
-                                 (or (= type constants/community-token-permission-can-view-channel)
-                                     (=
-                                      type
-                                      constants/community-token-permission-can-view-and-post-channel))))]
-       {:db (assoc-in db [:communities id] community)
-        :fx [[:dispatch [:communities/initialize-permission-addresses id]]
-             (when (not joined)
-               [:dispatch [:chat.ui/spectate-community id]])
-             (when (nil? token-permissions-check)
-               [:dispatch [:communities/check-permissions-to-join-community id]])
-             (when (some has-channel-perm? token-permissions)
-               [:dispatch [:communities/check-all-community-channels-permissions id]])]}))))
+(defn <-revealed-accounts-rpc
```

Comment:
To follow our current standards, this namespace already contains logic that should be moved to the data store namespaces, particularly the `*-rpc` functions. It would be nice if you could write the new functions already in the data store directory, such as `<-revealed-accounts-rpc`.

Reply:
moved all these <- fn to legacy.status-im.data-store.communities